static-lib.mk

```diff
@@ -0,0 +1,70 @@
+# Makefile rules for compiling and static library
+# supported external variables:
+# - NAME - component/target binary name
+# - LOCAL_SRCS - list of source files relative to current makefile
+# - SRCS       - list of source files relative to project root
+# - LOCAL_HEADERS - headers to be installed (relative to current makefile)
+# - HEADERS       - headers to be installed (relative to project root)
+#
+# - DEPS - list of components from current repo to be completed before starting this one
+#
+
+### FIXME: should be available in phoenix-rtos-build for usage in all projects
+
+# directory with current Makefile - relative to the repository root
+CLIENT_MAKES := $(filter-out $(binary.mk) $(static-lib.mk) %.c.d,$(MAKEFILE_LIST))
```

Comment:
`static-lib.mk` and `binary.mk` share much common code, I'm thinking about introducing some common files for them, but that's just implementation detail